Can't publish an integration with a dot in the name

## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description
I tried to publish an integration with a dot (`.`) in the name, got an error:
```
2018-10-22 10:32:03.355 ERROR [-,,,] 1 --- [tion Controller] i.s.s.c.integration.online.BaseHandler   : Integration [test-no-basepath-api.json]: [ERROR] Activation failure
javax.validation.ConstraintViolationException: Constraint Validations: name must match "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$" on bean: Container(args=[], command=[], env=[EnvVar(name=LOADER_HOME, value=${JAVA_DATA_DIR}/syndesis/loader, valueFrom=null, additionalProperties={}), EnvVar(name=AB_JMX_EXPORTER_CONFIG, value=/tmp/src/prometheus-config.yml, valueFrom=null, additionalProperties={})], envFrom=[], image=172.30.1.1:5000/syndesis/i-test-no-basepath-api.json:1, imagePullPolicy=Always, lifecycle=null, livenessProbe=null, name=i-test-no-basepath-api.json, ports=[ContainerPort(containerPort=8778, hostIP=null, hostPort=null, name=jolokia, protocol=null, additionalProperties={})], readinessProbe=null, resources=null, securityContext=null, stdin=null, stdinOnce=null, terminationMessagePath=null, terminationMessagePolicy=null, tty=null, volumeMounts=[VolumeMount(mountPath=/deployments/config, name=secret-volume, readOnly=false, subPath=null, additionalProperties={})], workingDir=null, additionalProperties={})
	at io.fabric8.kubernetes.api.builder.ValidationUtils.validate(ValidationUtils.java:36) ~[kubernetes-model-2.0.10.jar!/:2.0.10]
	at io.fabric8.kubernetes.api.model.ContainerBuilder.build(ContainerBuilder.java:81) ~[kubernetes-model-2.0.10.jar!/:2.0.10]
	at io.fabric8.kubernetes.api.model.PodSpecFluentImpl$ContainersNestedImpl.and(PodSpecFluentImpl.java:1063) ~[kubernetes-model-2.0.10.jar!/:2.0.10]
	at io.fabric8.kubernetes.api.model.PodSpecFluentImpl$ContainersNestedImpl.endContainer(PodSpecFluentImpl.java:1066) ~[kubernetes-model-2.0.10.jar!/:2.0.10]
	at io.syndesis.server.openshift.OpenShiftServiceImpl.ensureDeploymentConfig(OpenShiftServiceImpl.java:253) ~[server-openshift-1.5-SNAPSHOT.jar!/:1.5-SNAPSHOT]
	at io.syndesis.server.openshift.OpenShiftServiceImpl.deploy(OpenShiftServiceImpl.java:87) ~[server-openshift-1.5-SNAPSHOT.jar!/:1.5-SNAPSHOT]
	at io.syndesis.server.controller.integration.online.PublishHandler.deploy(PublishHandler.java:182) ~[server-controller-1.5-SNAPSHOT.jar!/:1.5-SNAPSHOT]
	at io.syndesis.server.controller.integration.online.PublishHandler$BuildStepPerformer.perform(PublishHandler.java:333) ~[server-controller-1.5-SNAPSHOT.jar!/:1.5-SNAPSHOT]
	at io.syndesis.server.controller.integration.online.PublishHandler.execute(PublishHandler.java:115) ~[server-controller-1.5-SNAPSHOT.jar!/:1.5-SNAPSHOT]
	at io.syndesis.server.controller.StateChangeHandler.execute(StateChangeHandler.java:33) [server-controller-1.5-SNAPSHOT.jar!/:1.5-SNAPSHOT]
	at io.syndesis.server.controller.integration.IntegrationController.lambda$callStateChangeHandler$11(IntegrationController.java:199) [server-controller-1.5-SNAPSHOT.jar!/:1.5-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_151]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_151]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_151]
```

--- a/app/common/util/src/main/java/io/syndesis/common/util/Names.java
+++ b/app/common/util/src/main/java/io/syndesis/common/util/Names.java
@@ -21,12 +21,13 @@ import static io.syndesis.common.util.Strings.truncate;
 
 public final class Names {
 
-    private static final Pattern VALID_REGEX = Pattern.compile("^[a-z0-9][-A-Za-z0-9\\\\.]+$");
+    private static final Pattern VALID_REGEX = Pattern.compile("^[a-z0-9][-A-Za-z0-9\\\\]+$");
     private static final Pattern INVALID_START_REGEX = Pattern.compile("^[^a-z0-9]+");
     private static final Pattern INVALID_CHARACTER_REGEX = Pattern.compile("[^a-zA-Z0-9-\\._]");
     private static final String SPACE = " ";
     private static final String BLANK = "";
     private static final String DASH = "-";
+    private static final String DOT = "\\.";
     private static final String UNDERSCORE = "_";
     //The actual limit is 253,
     // but individual resources have other limitations
@@ -41,9 +42,10 @@ public final class Names {
      * Sanitizes the specified name by applying the following rules:
      * 1. Replace spaces with dashes.
      * 2. Replace underscores with dashes.
-     * 3. Ensures that the first character is a alphanumeric
-     * 4. Remove invalid characters.
-     * 5. Keep the first 64 characters.
+     * 3. Replace dots with dashes
+     * 4. Ensures that the first character is a alphanumeric
+     * 5. Remove invalid characters.
+     * 6. Keep the first 64 characters.
      * @param name  The specified name.
      * @return      The sanitized string.
      */
@@ -51,6 +53,7 @@ public final class Names {
         final String firstPass = name
             .replaceAll(SPACE, DASH)
             .replaceAll(UNDERSCORE, DASH)
+            .replaceAll(DOT, DASH)
             .toLowerCase(Locale.US);
 
         final String secondPass = INVALID_START_REGEX.matcher(firstPass).replaceAll(BLANK);

--- a/app/common/util/src/test/java/io/syndesis/common/util/NamesTest.java
+++ b/app/common/util/src/test/java/io/syndesis/common/util/NamesTest.java
@@ -55,7 +55,8 @@ public class NamesTest {
             pair("twitter-mention-salesforce-upsert-contact-",
                 "??? Twitter Mention <-> Salesforce upsert contact !!!"),
             pair("s-p-a-c-e-the-final-frontier", "    s   p  a  c  e   , the final frontier"),
-            pair("walking-on-sunshine-", "_walking_on_sunshine_"));
+            pair("walking-on-sunshine-", "_walking_on_sunshine_"),
+            pair("dot-", "dot."));
     }
 
     private static Object[] pair(final String projectName, final String integrationName) {

